scripts/cangen/cangen/util.py

```diff
@@ -0,0 +1,22 @@
+import re
+
+
+def _camel_to_snake(text):
+    """Converts UpperCamelCase to snake_case."""
+
+    s1 = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", text)
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", s1).lower()
+
+
+def _decimal_to_hex(decimal_value):
```

Comment:
We should get rid of this function and just use `hex` directly. It's an unnecessary abstraction (ik you didn't write this, but we should clean it out)